src/context.c

```diff
@@ -130,6 +138,73 @@ static void set_ub_edns_maximum_udp_payload_size(struct getdns_context*,
 /* Stuff to make it compile pedantically */
 #define RETURN_IF_NULL(ptr, code) if(ptr == NULL) return code;
 
+
+#ifdef USE_WINSOCK
+// For windows, the CA trust store is not read by openssl. 
+// Add code to open the trust store using wincrypt API and add
+// the root certs into openssl trust store
+static int 
+add_WIN_cacerts_to_openssl_store(SSL_CTX* tls_ctx)
+{
+	HCERTSTORE      hSystemStore;
+	PCCERT_CONTEXT  pTargetCert = NULL;
+
+	// load just once per context lifetime for this version of getdns
+	// TODO: dynamically update CA trust changes as they are available
+	if (!tls_ctx)
+		return 0;
+
+	// Call wincrypt's CertOpenStore to open the CA root store.
+
+	if ((hSystemStore = CertOpenStore(
+		CERT_STORE_PROV_SYSTEM,
+		0,
+		NULL,
+		// NOTE: mingw does not have this const: replace with 1 << 16 from code
+		// CERT_SYSTEM_STORE_CURRENT_USER,
+		1 << 16,
+		L"root")) == 0)
+	{
+		return 0;
+	}
+
+	X509_STORE* store = SSL_CTX_get_cert_store(tls_ctx);
+	if (!store)
+		return 0;
+
+	// iterate over the windows cert store and add to openssl store
+	while (pTargetCert = CertEnumCertificatesInStore(
+		hSystemStore,
+		pTargetCert))
+	{
+		X509 *cert1 = d2i_X509(NULL, (const unsigned char **)&pTargetCert->pbCertEncoded, pTargetCert->cbCertEncoded);
+		if (!cert1) 
+			// do not return if a cert fails, continue and retrieve the rest
+			DEBUG_STUB("*** %s(%s)\n", __FUNCTION__,
+			"unable to parse certificate in memory");

```

Comment:
This path and the one below with the DEBUG_STUB call both need to exit as error paths so please add a "return 0;" statement back in.